src/ngraph/node.cpp

```diff
@@ -450,7 +431,7 @@ std::shared_ptr<Node> Node::get_argument(size_t index) const
 {
     NGRAPH_CHECK(
         index < m_inputs.size(), "index '", index, "' out of range in get_argument(size_t index)");
-    return input_value(index).as_single_output_node();
+    return input_value(index).get_node_shared_ptr();
```

Comment:
I suspect most remaining callers assume the argument is a one-output node and this should really be returning an `Output<Node>`.

Reply:
Most uses actually want the node, not the Output<Node>. A very common use is the cast the input argument to a particular op type. There are some misuses of get_arguments, but certainly not the majority.

Reply:
Many of the incorrect uses have already been changed; the replacement approach was replacing getArgument users with something else. It used to be that most uses were incorrect.